devops/deploy/supervisor.py

```diff
@@ -0,0 +1,33 @@
+# Importing
+import asyncio
+
+# Placeholder for function to recover nodes recover_node(host, port)
+
+
+# Function to check if a node is alive
+async def ping_node(host, port):
+    try:
+        reader, writer = await asyncio.open_connection(host, port)
+        print(f"Alive node at {host}:{port}")
```

Comment:
A little more on this beyond the above and our conversation.

In a deployment environment for something we want to ensure stays up, we'd usually use (in increasing preference and not an exhaustive list) one of:
1. a monitor script with `cron`
2. a supervising tool such as `daemontools` or `supervisord`
3. `systemd`

Determining _"liveness"_ of a service via a network call is somewhat fragile since the network path from the monitoring app could have a partition, but other apps can reach it just fine. The more degenerate case is that the monitoring app can reach it, but the apps that depend on it can't. A more refined approach is to look at client app calls for signs of health (assuming a reasonable degree of traffic).